### PR TITLE
DSP-788 Fix compatibility issues with CKEditor version 4

### PR DESCRIPTION
**Pull request by tobiasschweizer**
resolves [DSP-788](https://dasch.myjetbrains.com/youtrack/issue/DSP-788)

### CODE REVIEW

File: projects/dsp-ui/src/lib/viewer/values/text-value/text-value-as-xml/text-value-as-xml.component.ts [hunk 1]
```diff
@@ -90,12 +91,14 @@ export class TextValueAsXMLComponent extends BaseValueComponent implements OnIni
...
-                    {model: 'heading5', view: 'h5', title: 'Heading 5'},
-                    {model: 'heading6', view: 'h6', title: 'Heading 6'},
+                    { model: 'paragraph', title: 'Paragraph', class: 'ck-heading_paragraph' },
+                    { model: 'heading1', view: 'h1', title: 'Heading 1' },
+                    { model: 'heading2', view: 'h2', title: 'Heading 2' },
+                    { model: 'heading3', view: 'h3', title: 'Heading 3' },
+                    { model: 'heading4', view: 'h4', title: 'Heading 4' },
+                    { model: 'heading5', view: 'h5', title: 'Heading 5' },
+                    { model: 'heading6', view: 'h6', title: 'Heading 6' },
+                    { model: 'formatted', view: 'pre', title: 'Formatted' }
```
Comment (tobiasschweizer) [inline projects/dsp-ui/src/lib/viewer/values/text-value/text-value-as-xml/text-value-as-xml.component.ts:101]: @loicjaouen I could add support for the `<pre>` element. However, according to the docs it is not available in [CKeditor 5](https://ckeditor.com/docs/ckeditor5/latest/builds/guides/migrate.html#configuration-options-compatibility-table):

Screenshot 2020-10-23 at 13 42 02

Comment (tobiasschweizer) [reply projects/dsp-ui/src/lib/viewer/values/text-value/text-value-as-xml/text-value-as-xml.component.ts:101]: I posted the question on stackoverflow: https://stackoverflow.com/questions/64499528/support-of-element-pre-in-ckeditor-5

**Comment by tobiasschweizer** [comment]
I think one can use custom formatting options: https://ckeditor.com/docs/ckeditor5/latest/features/headings.html

**Comment by loicjaouen** [comment]
would it be possible to have a read-only css that looks closer to the edit version?
two-css

**Comment by tobiasschweizer** [comment]
@loicjaouen I checked an saw that `<pre>` and codeblock share common CSS:

Screenshot 2020-10-26 at 08 48 58

```css
.ck-content pre {
    padding: 1em;
    color: #353535;
    background: hsla(0,0%,78%,.3);
    border: 1px solid #c4c4c4;
    border-radius: 2px;
    text-align: left;
    direction: ltr;
    tab-size: 4;
    white-space: pre-wrap;
    font-style: normal;
    min-width: 200px;
}

```

The back ground color could be removed, but then this would also affect code block.

**Comment by tobiasschweizer** [comment]
@loicjaouen I figured that `<cite>` can be supported via the selection (heading etc.). I could also add `<pre>` as "Formatted".

Screenshot 2020-10-26 at 08 53 38

**Comment by tobiasschweizer** [comment]
Markup Matrix:

CKEditor 4 to 5 migration: https://ckeditor.com/docs/ckeditor5/latest/builds/guides/migrate.html

Differences:
- italics: `<em>` -> `<i>` 
- strike: `<strike>` -> `<s>`
- table 
```html
  <table>
        <tr>
            <td>year</td>
            <td>name</td>
        </tr>
        <tr>
            <td>2012</td>
            <td>Old Ideas</td>
        </tr>
        <tr>
            <td>2014</td>
            <td>Polpular Problems</td>
        </tr>
    </table>
  ```
 ```html
    <figure class="table">
        <table>
           <tbody>
               <tr>
                 <td></td>
               </tr>
            </tbody>
      </table>
   ```
- codeBlock: `<pre>` -> `<pre><code class="language-css"></code></pre>`

Identical:
- paragraph: `<p>`
- bold: `<strong>`
- underline: `<u>`
- subscript: `<sub>`
- superscript: `<sup>`
- external link: `<a>` with `href` (no class)
- horizontal line: `<hr>`
- unordered list: `<ul><li></li></ul>` 
- ordered list: `<ol><li></li></ol>`
- blok quote: `<blockquote>`

Not implemented yet:
- ~~internal link with class "salsah-link"~~
- ~~`<h1>` - `<h6>`~~
- ~~`<code>`~~
- ~~`<cite>`~~
- ~~`<pre>`~~
- ~~remove `&nbsp;`~~

**Comment by tobiasschweizer** [comment]
@loicjaouen I removed the background color for `<pre>` (formatted text, code block) in https://github.com/dasch-swiss/dsp-ui-lib/pull/212/commits/e324b0658789d8a7e16f6a0d7015384273fd6204

**Comment by tobiasschweizer** [comment]
@loicjaouen Did you find any problems relating to unsupported elements?

**Comment by tobiasschweizer** [comment]
@loicjaouen This PR has been replaced by #224

**Pull request merged**